server/src/main/java/io/crate/analyze/TableElementsAnalyzer.java

```diff
@@ -408,6 +410,15 @@ public AnalyzedAlterTableAddColumn analyze(AlterTableAddColumn<Expression> alter
         return new AnalyzedAlterTableAddColumn(table, columns, checks);
     }
 
+    public AnalyzedAlterTableDropColumn analyze(AlterTableDropColumn<Expression> alterTable) {
```

Comment:
I'd move this to `AlterTableDropColumnAnalyzer`. The `TableElementsAnalyzer` is intended for new columns. If the columns exist we can use a regular ExpressionAnalyzer with `TableReferenceResolver`.

Given that the drop column statement also doesn't take any parameters we can resolve `Reference` directly in the analyze phase, and don't have to bind/eval at all.

The dispatching via a visitor is also not necessary as the `DropColumnDefinition` contains all info and has no further derived classes.

Reply:
Thx! I chose to follow the new code to be consistent but you're right it's completely unnecessary, opened PR to simplify: https://github.com/crate/crate/pull/14547